tests/UrlGeneratorTest.php

```diff
@@ -183,4 +193,120 @@ public function testDefaultNotUsedForRequiredParameter(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromRouteHostParamWithSchemeParamGenerated(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index', [], 'https');
+
+        $this->assertEquals('https://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromRouteHostParamWithTrailingSlashGenerated(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com/'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromCurrentHostGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index'),
+        ];
+
+        $router = $this->createUrlGenerator($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromCurrentHostWithNonStandardPortGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com:8080/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index'),
+        ];
+
+        $router = $this->createRouter($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com:8080/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromUrlHostParamWithProtocolRelativeSchemeGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index')->host('//test.com'),
+        ];
+
+        $router = $this->createRouter($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromMethodHostParamWithProtocolRelativeSchemeGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index')->host('//mysite.com'),
+        ];
+
+        $router = $this->createRouter($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index', [], null, '//test.com');
+
+        $this->assertEquals('http://test.com/home/index', $url);
```

Comment:
Same here. I'd not expect protocol being changed to `http`.

Reply:
> Same here. I'd not expect protocol being changed to `http`.

That's what yii2 does.
You can avoid this by setting the schema to an empty string.
```php
$router->generateAbsolute('index', [], '', '//test.com');
```

Reply:
I think the removed test for overriding a scheme of the $host method param should be.

Reply:
> That's what yii2 does.

Doesn't mean it's correct and expected...

Reply:
> I think the removed test for overriding the scheme of the $host method param should be.
Tests should reflect every use case.

It was exact duplicate of another test.

Reply:
> It was exact duplicate of another test.

Which one?

Reply:
> > That's what yii2 does.
> 
> Doesn't mean it's correct and expected...

Ok, just imagine we fix this behavior, in what way I can override the scheme by the current scheme?

Reply:
I propose the next syntax 
```php
//means protocol-relative links remain protocol-relative, other will be overriden by current scheme
$router->generateAbsolute('post-index', [], '//', '//test.com');
```

Reply:
Correct behavior is checking the following:

1. If `generateAbsolute()` `$scheme` parameter is specified, use that scheme
2. If there's a scheme in `generateAbsolute()` `$host`, use that scheme
3. If there's a host in route with a scheme, use that scheme
4. If there's current request available, use its scheme

Reply:
> Which one?

OK, maybe it was not :) Sorry. Anyway, the points above should be ensured.

Reply:
> Correct behavior is checking the following:
> 
> 1. If `generateAbsolute()` `$scheme` parameter is specified, use that scheme
> 2. If there's a scheme in `generateAbsolute()` `$host`, use that scheme
> 3. If there's a host in route with a scheme, use that scheme
> 4. If there's current request available, use its scheme

Maybe it makes sense, I need to think it over.